src/frameworks/next/index.ts

```diff
@@ -344,25 +344,26 @@ export async function ɵcodegenFunctionsDirectory(sourceDir: string, destDir: st
   const { distDir } = await getConfig(sourceDir);
   const packageJson = await readJSON(join(sourceDir, "package.json"));
   if (existsSync(join(sourceDir, "next.config.js"))) {
-    let esbuild;
-    try {
-      esbuild = await import("esbuild");
-    } catch (e: unknown) {
-      // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
-      logger.debug(`Failed to load 'esbuild': ${e}`);
-      throw new FirebaseError(
-        `Unable to find 'esbuild'. Install it into your local dev dependencies with 'npm i --save-dev esbuild''`
-      );
-    }
-    await esbuild.build({
-      bundle: true,
-      external: Object.keys(packageJson.dependencies),
-      absWorkingDir: sourceDir,
-      entryPoints: ["next.config.js"],
-      outfile: join(destDir, "next.config.js"),
-      target: `node${NODE_VERSION}`,
-      platform: "node",
-    });
+    // Bundle their next.config.js with esbuild via NPX, pinned version was having troubles on m1
+    // macs and older Node versions; either way, we should avoid taking on any deps in firebase-tools
+    // Alternatively I tried using @swc/spack and the webpack bundled into Next.js but was
+    // encountering difficulties with both of those
+    const dependencyTree: NpmLsReturn = JSON.parse(
+      execSync(`npm ls --omit=dev --all --json`, { cwd: sourceDir }).toString()
```

Comment:
Does this preclude `pNPM` or other package managers? (I honestly don;t know)

Reply:
`npm ls` should still return correct results regardless of the package manager, just confirmed with a codebase of mine that uses yarn